### PR TITLE
Complete draft of protobuf post

### PR DESCRIPTION
**Pull request by rebeccabilbro**
This post is intended to be a gentle introduction to protocol buffers

### CODE REVIEW

**Review by bbengfort — APPROVED** [review]
Excellent post! I think this is a good description of serialization and deserialization, particularly for an audience of folks who are shifting from thinking about the experimental/analytical side of data science to the operational side (e.g. building APIs for models). I think that's a wonderful target audience, particularly for folks we're thinking about bringing on board to work with us. I also really liked the idea of matching your serialization format to your use case - I think that's an important comment to make; protocol buffers are great for microservices but are probably not the right match for data APIs as you suggested. 

I've pushed some minor edits to the document and made some other comments for you to review. My big comment though is that I thought that the transition from a data science-focused audience to protobufs designed around transactions was a bit distracting. I propose that we continue on theme by doing a protocol buffer definition for a model management system API. This might tie in a bit better with the earlier code snippets loading/dumping pickles. 

Here is my suggested protobuf with comments, along with some example Python code:

https://gist.github.com/bbengfort/f2460ecc01f8b8b9d7861ab33f50488a

I realize this is a big shift and requires rewriting the second section, so if you don't think it's a good idea - no worries! 

Thank you so much for writing this blog post! 


https://gist.github.com/bbengfort/f2460ecc01f8b8b9d7861ab33f50488a

File: content/en/blog/2021-05-19-what-are-protocol-buffers.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+```
+
+... will generate a new file, `transfer.pb.go` that contains Golang code that can be used to write to and read from the schema we defined in `transfer.proto`. If we wanted to compile the code to Python, we'd do:
+
+```bash
+protoc --python_out=. transfer.proto
+```
+... which would generate `transfer_pb2.py`.
+
+## Final Thoughts
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply content/en/blog/2021-05-19-what-are-protocol-buffers.md:121]: I included an example in the gist if it's useful.

**Comment by rebeccabilbro** [comment]
Thank you for your thorough review and thoughtful comments @bbengfort &mdash; that's a really good point about keeping the audience and context more consistent throughout the post. 

I'm sorry this ended up being more of a lift for you than I meant it to be, particularly on such a busy week, but the feedback is very valuable, and I think it's worth the work to get this one right. My hope is, as you suggested, that this post can become a helpful reference point for people who are transitioning from data science to distributed systems, since this will likely describe many of our future teammates.

Maybe it's better not to use the transaction example; it felt relevant to me at the time given that everyone is thinking about crypto these days, but perhaps it's too close to what we've done for a recent contract (though I did attempt to genericize it). 

I'll take a crack at rewriting the post tonight and hopefully can get it there without too much delay. The code you wrote and have provided in the gists will certainly help; thank you for doing that as well.

**Review by bbengfort — APPROVED** [review]
Thank you for taking the time to do a second pass, if the first post was good, this revision is excellent! I think the structure flows well now and I really like how you broke up the use cases and gave them headings, it highlighted their importance and drove home the main point of the article. Nicely done!

**Pull request merged**